analyses/cell-type-consensus/scripts/04-combine-celltype-tables.R

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env Rscript
+
+# This script is used to combine all TSV files containing cell types into a single TSV file 
+# The output TSV file will include the following added columns: 
+# panglao_ontology: CL term assigned to panglao term
+# panglao_annotation: human readable value associated with the CL term for panglao term 
+# blueprint_annotation_fine: Fine-grained annotation from blueprint associated with singler_celltype_ontology 
+# consensus_annotation: human readable name associated with the consensus label 
+# consensus_ontology: CL ontology term for the consensus cell type 
+
+project_root <- here::here()
+
+library(optparse)
+
+option_list <- list(
+  make_option(
+    opt_str = c("--celltype_tsv_dir"),
+    type = "character",
+    help = "Path to directory containing TSV files with cell type annotations from single samples.
+      All TSV files in this directory will be combined into a single file."
+  ),
+  make_option(
+    opt_str = c("--panglao_ref_file"),
+    default = file.path(project_root, "references", "panglao-cell-type-ontologies.tsv"),
+    type = "character", 
+    help = "Path to file with panglao assignments and associated cell ontology ids"
+  ),
+  make_option(
+    opt_str = c("--consensus_ref_file"),
+    default = file.path(project_root, "references", "consensus-cell-type-reference.tsv"),
+    type = "character",
+    help = "Path to file containing the reference for assigning consensus cell type labels"
+  ), 
+  make_option(
+    opt_str = c("--output_file"),
+    type = "character",
+    help = "Path to file where combined TSV file will be saved, must end in `.tsv`"
+  )
```

Comment:
I assume the output file here is large: do we want to allow compressed file output? You don't have code to enforce the `.tsv` ending, and `write_tsv` supports this just by specifying the compression in file name (`.tsv.gz`, etc.) so just updating the help text would be sufficient here.